ff_module-form-preview react

sd-92 opened the pull request.

sd-92 committed:
Merge branch 'dev' into sd-ff_module-form-preview-react

sd-92 committed:
Added nice inner-shadow

sd-92 committed:
form-preview to react

sd-92 committed:
Added tests

sd-92 committed:
Merge branch 'dev' into sd-ff_module-form-preview-react

sd-92 committed:
added inline-edit module and updated tests

dkhuntrods commented on blocks/core/ff_module/ff_module-form-preview/ff_module-form-preview.md:
```diff
@@ -25,3 +25,6 @@ data:
 ---
+## React Component
+
+<div data-ff_module-form-preview="" ></div>
```
Can you add a bit of detail to the md file regarding proptypes (see `ff_module-dropdown-button-component` for examples), and maybe some code-formatted flat html to have an example (I've started adding these in my modal-dialog branch if you want to see what I mean) as this one is not so straightforward layout-wise.

dkhuntrods commented on blocks/core/ff_module/ff_module-form-preview/ff_module-form-preview.js:
```diff
@@ -0,0 +1,36 @@
+var InlineEdit = require('../ff_module-inline-edit/ff_module-inline-edit');
+
+module.exports = React.createClass({
+	displayName: 'FormPreview',
```
Can you add a propTypes declaration? (See ff_module-dropdown-button-component and https://facebook.github.io/react/docs/reusable-components.html)

sd-92 committed:
Added PropTypes and more info into md file

dkhuntrods commented on blocks/core/ff_module/ff_module-form-preview/ff_module-form-preview.js:
```diff
@@ -0,0 +1,45 @@
+	displayName: 'FormPreview',
+	propTypes: {
+		items: React.PropTypes.array.isRequired,
+		title: React.PropTypes.string,
```
Everything here and below is internal to each `item` in the `items` array, isn't it? So it's not a propType of this component, right?

sd-92 committed:
Merge branch 'dev' into sd-ff_module-form-preview-react

sd-92 committed:
Updated propTypes.

sd-92 committed:
Added required fields

To make the simplest element only a title is required. The value could
be omitted and substituted with a list or html
